SERVLET4_PUSH_TESTS-21780 #21780

edburns opened the pull request:
Tests for issue #21780 

	modified:   .gitignore
	modified:   main/appserver/tests/appserv-tests/devtests/web/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/index.html
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/my.css
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/TestServlet.java

- Tests from before change to github.

Commit:
SERVLET4_PUSH_TESTS-21780 #21780

	modified:   .gitignore
	modified:   main/appserver/tests/appserv-tests/devtests/web/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/index.html
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/my.css
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/TestServlet.java

- Tests from before change to github.

Commit:
SERVLET4_PUSH_TESTS-21780 #21780

	modified:   .gitignore
	modified:   main/appserver/tests/appserv-tests/devtests/web/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/index.html
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/my.css
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/TestServlet.java

- Tests from before change to github.

Commit:
SERVLET4_PUSH_TESTS-21780 #21780

	modified:   .gitignore
	modified:   main/appserver/tests/appserv-tests/devtests/web/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/index.html
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/my.css
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/TestServlet.java

- Tests from before change to github.

Commit:
SERVLET4_PUSH_TESTS-21780 #21780

	modified:   .gitignore
	modified:   main/appserver/tests/appserv-tests/devtests/web/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/index.html
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/my.css
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/TestServlet.java

- Tests from before change to github.

Commit:
SERVLET4_PUSH_TESTS-21780 #21780

	modified:   .gitignore
	modified:   main/appserver/tests/appserv-tests/devtests/web/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/index.html
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/docroot/my.css
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java
	new file:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/TestServlet.java

- Tests from before change to github.

edburns commented:
@glassfishrobot Run CI tests please

edburns commented:
http://gf-hudson.us.oracle.com/hudson/job/gf-remote-queue/344/

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java:
```diff
@@ -0,0 +1,105 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2012-2017 Oracle and/or its affiliates. All rights reserved.
```
This should just be 2017 as it's a new file.

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml:
```diff
@@ -0,0 +1,115 @@
+
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+
+    Copyright (c) 1997-2017 Oracle and/or its affiliates. All rights reserved.
```
The year should just be 2017.

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java:
```diff
@@ -0,0 +1,80 @@
+            String pathInfo = req.getPathInfo();
+            String method = req.getMethod();
+            if ("FOO".equals(method)) {
+                    System.out.println("debug: edburns: sending OK: pathInfo:" + pathInfo);
```
Probably want to remove this...

edburns replied:
Thanks, removed it.

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java:
```diff
@@ -0,0 +1,107 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2012-2017 Oracle and/or its affiliates. All rights reserved.
```
Copyright year.

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml:
```diff
@@ -0,0 +1,115 @@
+
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+
+    Copyright (c) 1997-2017 Oracle and/or its affiliates. All rights reserved.
```
Copyright year.

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java:
```diff
@@ -0,0 +1,80 @@
+                    if ("/my.css".equals(pathInfo)) {
+                            res.getWriter().println("foo: bar");
+                            res.setStatus(HttpServletResponse.SC_OK);
+
```
Shouldn't this return after setting the status?

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java:
```diff
@@ -0,0 +1,80 @@
+            }
+            if ("/my.css".equals(pathInfo)) {
+                    res.getWriter().println("foo: bar");
+                    res.setStatus(HttpServletResponse.SC_OK);
```
shouldn't this return after setting the status?

edburns replied:
Fixed.

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java:
```diff
@@ -0,0 +1,87 @@
+                    if (null != (result = req.getParameter("ln"))) {
+                            if (!"resources".equals(result)) {
+                                    req.getServletContext().log("Parameter value incorrect.  Should be resources.  Is: " + result);
+                                    res.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```
Shouldn't a return be here after setting the status?

rlubke commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java:
```diff
@@ -0,0 +1,87 @@
+                    } else {
+                            req.getServletContext().log("Should have ln parameter.  Does not.");
+                            res.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                            
```
Missing return?

edburns replied:
Fixed.

Commit:
On branch SERVLET4_PUSH_TESTS-21780

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml

- copyright

modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/ser
vlet/test/FooMethodServlet.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java

- add return

Commit:
On branch SERVLET4_PUSH_TESTS-21780

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml

- copyright

modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/ser
vlet/test/FooMethodServlet.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java

- add return

Commit:
On branch SERVLET4_PUSH_TESTS-21780

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.xml
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.xml

- copyright

modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/ser
vlet/test/FooMethodServlet.java
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java

- add return

Commit:
fix conflicts

edburns commented:
@glassfishrobot run CI tests please

edburns commented:
http://gf-hudson.us.oracle.com/hudson/job/gf-remote-queue/356/

shingwaichan commented:
I have the following comments on the change:
1. pushArbitraryMethod/servlet/test/TestServlet.java does not write the body to invoke FOO foo/my.css. 
2. pushArbitraryMethod/WebTest.java has comments and description for "push static file"
3. pushDynamic/WebTest.java has comments and description for "push static file"

It would be good to fix that.

Commit:
SERVLET4_PUSH_TESTS-21780 Apply comments

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java

- Make comment match test.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java

- Remove debugging statement.

- Add returns.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java

- Write body and status code.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java

- Make comment match test.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java

- Add returns.

Commit:
SERVLET4_PUSH_TESTS-21780 Apply comments

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java

- Make comment match test.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/FooMethodServlet.java

- Remove debugging statement.

- Add returns.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java

- Write body and status code.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/WebTest.java

- Make comment match test.

	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/servlet/test/NOTfacesServlet.java

- Add returns.

Commit:
Merge conflicts

shingwaichan reviewed (changes requested):
1. pushArbitraryMethod/WebTest.java, the following description is not correct : 
    stat.addDescription("Unit test for http2 push static file");
2. pushArbitraryMethod/servlet/test/TestServlet.java, we have the following: 
res.getWriter().println("<html><head><title>HTTP2 Test</title><link rel=\"stylesheet\" href=\"/foo/my.css\"></head><body>Hello</body></html>"); 
    The above html body will not access “foo/my.css” through the FOO method.  
    In other words, the pushed “foo/my.css” will not be accessed by a browser.

Commit:
merge conflicts

Commit:
Address Shing-wai comments https://github.com/javaee/glassfish/pull/21786#pullrequestreview-42138448

edburns commented:
<img width="594" alt="20170605-glassfish-pr-21786-pusharbitrarymethod" src="https://cloud.githubusercontent.com/assets/75821/26803921/0d91112c-4a14-11e7-8d50-5e4f9c1e9356.png">

This image of chrome net-internals shows the test is effective in the browser as written.

edburns commented:
I assert that no further change is needed to the pushArbitraryMethod test and request this review be approved.  

Thanks,

Ed

shingwaichan commented:
The chrome net-internals shows the FOO request has arrived. My original change request is that the given http response body "need to" spawn a FOO request. (In other words, the client does not ask for the push promise here.)
Per discussion with Ed, he will update the WebTest.java to verify that the method of push promise is  "FOO" by using client API. And he will investigate the use of javascript to spawn FOO request issue later.

Commit:
Satisfy Shing-wai review requirements

modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java

- Update EXPECTED_BODY.

- Separate out the two calls to verify().

- Modify push inspection to assert the method is FOO.

modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/servlet/test/TestServlet.java

- Documented how to test this in the browser.

Commit:
Merge branch 'master' into SERVLET4_PUSH_TESTS-21780

edburns commented:
@glassfishrobot run ci tests please

Commit:
Merge upstream

Commit:
Fix copyright.  See FAQ question 17. What are the different features of this Remote Queue

edburns commented:
@glassfishrobot run CI tests please

arindam-bandyopadhyay commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/WebTest.java:
```diff
@@ -0,0 +1,115 @@
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
```
This is old copyright header.Please update the copyright to the latest copyright header.

edburns replied:
I removed the copyright statement and used the copyright-excludes.

arindam-bandyopadhyay commented on main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties:
```diff
@@ -0,0 +1,46 @@
+    and Distribution License("CDDL") (collectively, the "License").  You
+    may not use this file except in compliance with the License.  You can
+    obtain a copy of the License at
+    https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
```
This is old copyright header.Please update the copyright to the latest copyright header.

shingwaichan commented:
To be consistence, we should update main/copyright/src/main/resources/copyright-exclude
 for the copyright rather than adding build.properties in copyright header.
Please see d8c544cfba9bd0dd5fb6b90e5e6943a9bbac3d05 as an example.

Commit:
Merge branch 'master' into SERVLET4_PUSH_TESTS-21780

Commit:
modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushArbitraryMethod/build.properties
	modified:   main/appserver/tests/appserv-tests/devtests/web/servlet-4.0/pushDynamic/build.properties

- remove copyright

	modified:   main/copyright/src/main/resources/copyright-exclude

- Add build.properties files to exclude list

Commit:
At least sort the servlet-4.0 exclude files

edburns commented:
@Glassfishrobot run CI tests please
